src/components/menu-surface/menu-surface.tsx

```diff
@@ -106,12 +112,32 @@ export class MenuSurface {
     }
 
     private handleDocumentClick(event) {
-        if (this.open && !isDescendant(event.target, this.host)) {
+        const elementPath = event.path || [];
+
+        const clickedInAllowedElement = this.isElementInPath(
```

Comment:
We want to check `this.open` before we do this. At the moment, each menu-surface loops through the path for every single click, even when no surface-area is open. In the docs for `limel-select`, that's 10 times for each click 😅

We should also check if `this.allowClicksElement` is even truthy before we run this. If it's not, there's no reason to do the loop at all.

Reply:
I think we could also replace our own private method with something like t his :thinking: 
```ts
const clickedInAllowedElement = elementPath.includes(this.allowClicksElement);
```

Reply:
> I think we could also replace our own private method with something like this 
> 
> ```ts
> const clickedInAllowedElement = elementPath.includes(this.allowClicksElement);
> ```

ah yeah now we can do that!

Reply:
> We want to check `this.open` before we do this. At the moment, each menu-surface loops through the path for every single click, even when no surface-area is open. In the docs for `limel-select`, that's 10 times for each click 
> 
> We should also check if `this.allowClicksElement` is even truthy before we run this. If it's not, there's no reason to do the loop at all.

I love loops!!!